src/Test/StateMachine/Types.hs

```diff
@@ -105,8 +109,13 @@ type Postcondition model act = forall resp.
 --   check.
 type InitialModel m = forall (v :: * -> *). m v
 
+data Result resp err = Ok resp | Fail err
```

Comment:
Maybe have the arguments in opposite order? In case we want `Result` to be a a monad or whatever in the future?

Reply:
Good point! Will fix in next PR.